pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -150,7 +152,14 @@ type Transformer struct {
 	sourceIdentifier    entry.Field
 
 	sync.Mutex
-	batchMap map[string][]*entry.Entry
+	batchMap   map[string]*SourceBatch
+	maxLogSize int64
+}
+
+// SourceBatch contains the status info of a batch
+type SourceBatch struct {
```

Comment:
Let's not export this struct, to keep the publicly exposed API to a minimum.

Reply:
done